shared/js/intl_helper.js

```diff
@@ -0,0 +1,148 @@
+(function(global) {
+  'use strict';
+
+  /*************************
+
+  IntlHelper.defineObject('shortDate', 'datetime', {
+    year: 'numeric',
+    month: 'long',
+  });
+
+  function setValue() {
+    var formatter = IntlHelper.getObject('shortDate');
+    console.log(formatter.format(new Date()));
+  }
+
+  setValue();
+
+  IntlHelper.on('shortDate', setValue);
+  
+   ****************************/
+  var objectCache;
+
+  var knownObjects = [
+    'datetime',
+    'number',
+  ];
+
+  global.IntlHelper = {
+    defineObject: function(name, type, options) {
+      if (knownObjects.indexOf(type) === -1) {
+        throw new Error('Unknown type: ' + type);
+      }
+
+      if (!objectCache) {
+        objectCache = new Map();
+      }
+
+      if (objectCache.has(name)) {
+        throw new Error(
+          'Intl Object with name "' + name + '" already exists');
+      }
+
+      objectCache.set(name, {
+        options,
+        type,
+        object: undefined,
+        listeners: new Set()
+      });
+    },
+
+    getObject: function(name) {
+      if (!objectCache.has(name)) {
+        throw new Error('Intl Object with name "' + name + '" is not defined');
+      }
+      const obj = objectCache.get(name);
+
+      if (obj.object === undefined) {
+        switch (obj.type) {
+          case 'number':
+            obj.object = Intl.NumberFormat(
+              navigator.languages,
+              obj.options
+            );
+            break;
+          case 'datetime':
+            const options = Object.assign({}, obj.options);
+            if (options.hour) {
+              options.hour12 = navigator.mozHour12;
+            }
+            console.log('new Format');
+            console.log(options);
+            obj.object = Intl.DateTimeFormat(
+              navigator.languages,
+              options
+            );
+            break;
+          default:
+            throw new Error('Unknown type: ' + obj.type);
+        }
+      }
+      return obj.object;
+    },
+    on: function(names, cb) {
+      if (typeof names === 'string') {
+        names = [names];
```

Comment:
OK, I see your point. I'll leave this up to you, since I don't want to block.  My suggestion would be to start with a more strict API and see how it's used and only then potentially introduce the polymorphic array attribute.  It's easy to extend this API in this direction, but not the other way around.

It's a tiny detail though.